### ISSUE TITLE
🐛 BUG: named slots generate `undefined=""` attributes

### ISSUE THREAD

**Issue by stefanprobst** [snowpackjs/astro#1453]
### What package manager are you using?

yarn

### What operating system are you using?

linux

### Describe the Bug

when using a named slot, the slotted html element has a `undefined=""` attribute in the generated html.

### Steps to Reproduce

1. `npm init astro`

2. add `src/layouts/BaseLayout.astro`:

```html
<html>
<body>
  <main>
    <slot name="main" />
  </main>
</body>
</html>
```

3. add `src/pages/index.astro`:

```html
---
import BaseLayout from '../layouts/BaseLayout.astro'
---
<BaseLayout>
  <h1 slot="main">Hello, World!</h1>
</BaseLayout>
```

4. `npx prettier dist/index.html | grep h1`:

```
 <main><h1 undefined="">Hello, World!</h1></main>
```

**Comment by matthewp** [comment]
I'm not seeing that happen here: https://github.com/matthewp/astro-issue-1453

Can you try that and see if you can change it to make the bug appear? Is it possibly that prettier is causing this?

**Comment by matthewp** [comment]
Closing but happy to rebuild if you can provide a reproduction, thanks!

**Comment by stefanprobst** [comment]
@matthewp thanks for looking into this. can you check what astro version you tried this with? in your test repo, i am getting astro v0.13.2, which indeed does not have the error. however, when updating to v0.20.9 i actually *can* reproduce the error with that repo.

Steps to reproduce:
1. `git clone https://github.com/matthewp/astro-issue-1453 && cd astro-issue-1453`
2. `yarn`
3. ```
   yarn why astro
   
   yarn why v1.22.5
   [1/4] Why do we have the module "astro"...?
   [2/4] Initialising dependency graph...
   [3/4] Finding dependency...
   [4/4] Calculating file sizes...
   => Found "astro@0.13.12"
   info Has been hoisted to "astro"
   info This module exists because it's specified in "devDependencies".
   ```
4. `yarn add -D astro`
5. ```
   yarn why astro
   
   yarn why v1.22.5
   [1/4] Why do we have the module "astro"...?
   [2/4] Initialising dependency graph...
   [3/4] Finding dependency...
   [4/4] Calculating file sizes...
   => Found "astro@0.20.9"
   info Has been hoisted to "astro"
   info This module exists because it's specified in "devDependencies".
   ```
6. `yarn build`
7. ```
   cat dist/index.html
   
   <!DOCTYPE html><html><head></head><body><main><h1 undefined="">Testing</h1></main></body></html>
   ```

**Comment by jasikpark** [comment]
Reopening for the above reproduction

**Comment by matthewp** [comment]
Thanks! In that case I'm marking this as Accepted.

**Comment by stefanprobst** [comment]
just tried with `astro@next--compiler` and it looks like this is fixed there.

**Comment by matthewp** [comment]
Thanks for checking it out! Going to leave open until that branch is released in 0.21.

**Comment by natemoo-re** [comment]
Following up on Matthew's previous comment, the team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
